### DISCUSSION TITLE
SELECT statement returns empty array even though data exists.

### DISCUSSION THREAD

**Discussion by Electric-Dragon** [supabase/supabase#6860] [Questions]
### Original Query

This is my current `SELECT` query which returns `[]`:

```
const { data, error } = await supabase
            .from('series_popularity')
            .select('series!inner(id,title,cover,summary,genre1,genre2,creator:public_profile!series_creator_fkey(username))')
            .order('popularity_score', { ascending: false })
            .limit(10)
```
---

### Query to check if data is returned

When I perform a `select(*)` through the `js-client`, it returns data:
 ```
[{series_id: "e9ba8ab4-b019-4eb6-afc6-492529966c0b", popularity_score: 0.5}] (1) (popular.js, line 34)
```
---

### Querying foreign table

If I try to query the `foreign table` without an `inner join` with the following query, I get the following error:

#### Query

```
const { data, error } = await supabase
            .from('series_popularity')
            .select('series_id(id,title,cover,summary,genre1,genre2,creator:public_profile!series_creator_fkey(username))')
            .order('popularity_score', { ascending: false })
            .limit(10)
```

#### Error

This is `error.message`:

```
Could not embed because more than one relationship was found for 'series_popularity' and 'series_id'
```

This is `error.hint`:

```
[Log] Try changing 'series_id' to one of the following: 'adaptation_notifications!adaptation_notifications_target_series_fkey', 'chapters!chapters_seriesid_fkey', 'series!series_adaptation_fkey', 'series_follows!series_follows_target_series_fkey'. Find the desired relationship in the 'details' key. (popular.js, line 31)
```

This is `error.details`:

```
[Log] Array (4) (popular.js, line 31)
0 {relationship: "adaptation_notifications_target_series_fkey using …s_id) and adaptation_notifications(target_series)", embedding: "series_popularity with adaptation_notifications", cardinality: "one-to-many"}
1 {relationship: "chapters_seriesid_fkey using series_popularity(series_id) and chapters(seriesid)", embedding: "series_popularity with chapters", cardinality: "one-to-many"}
2 {relationship: "series_adaptation_fkey using series_popularity(series_id) and series(adaptation)", embedding: "series_popularity with series", cardinality: "one-to-many"}
3 {relationship: "series_follows_target_series_fkey using series_pop…rity(series_id) and series_follows(target_series)", embedding: "series_popularity with series_follows", cardinality: "one-to-many"}

Array Prototyp

```
---

### View Definition

```
drop view series_popularity;

create or replace view series_popularity as (

  select id as series_id, ((0.5 * get_series_follows(id)) + (0.1 * series_total_reads.total_read_count) + (0.4 * series_total_likes.count)) as popularity_score

  from series

  join series_total_reads on series.id = series_total_reads.series_id

  join series_total_likes on series.id = series_total_likes.seriesid

)
```

This view has data:
<img width="422" alt="Bildschirmfoto 2022-05-17 um 12 42 55 PM" src="https://user-images.githubusercontent.com/45729391/168751356-d961f5f9-957c-4b78-98db-c2c3d64cd22c.png">

**Comment by steve-chavez** [comment]
> If I try to query the foreign table without an inner join with the following query, I get the following error:
> Could not embed because more than one relationship was found for 'series_popularity' and 'series_id'

Note that this is not the same query as the one you're showing above, which uses `series` instead of `series_id`.

Try it as:

```js
const { data, error } = await supabase
            .from('series_popularity')
            .select('series(id,title,cover,summary,genre1,genre2,creator:public_profile!series_creator_fkey(username))')
            .order('popularity_score', { ascending: false })
            .limit(10)
```

**Reply by Electric-Dragon** [reply]
@steve-chavez  `series` is the table name, and `series_id` is the column name. I was directly using the `foreign table name` because it worked when I had the problem the first time.

When I run the query as per your code...it still returns an empty array

**Reply by Electric-Dragon** [reply]
Hi @steve-chavez any help would be appreciated

**Reply by steve-chavez** [reply]
Hm, it's just that `inner` filters rows based on the "inner join" between `series_popularity` and `series`, omitting `inner` should return data.

The other thing that occurs to me is that perhaps you have an RLS policy filtering the data? Maybe you could try with RLS disabled.

**Reply by Electric-Dragon** [reply]
@steve-chavez The table and view do not have `RLS policies` yet...so I'm not sure why the data is not being returned. And it did not return anything even after omitting `inner`.

**Reply by Electric-Dragon** [reply]
@steve-chavez any updates?

**Reply by steve-chavez** [reply]
@Electric-Dragon Sorry, this one fell off my radar :disappointed:. Did you manage to solve your issue? I'd have to go into your instance to see what could be wrong but I think it's related to the data, probably a filter is removing all the entries.

**Reply by Electric-Dragon** [reply]
Hi I decided to go with 2 queries inorder to make it work. One that gets the ids...and then the order one that selects from the other table according to the ids. Not the best solution but it works😅

**Reply by thiagopetherson** [reply]
> Hm, it's just that `inner` filters rows based on the "inner join" between `series_popularity` and `series`, omitting `inner` should return data.
> 
> The other thing that occurs to me is that perhaps you have an RLS policy filtering the data? Maybe you could try with RLS disabled.

It worked for me. Why Does It happened ?
